scale/storage/configuration/workspace_configuration.py

```diff
@@ -0,0 +1,71 @@
+"""Defines the configuration for a storage Workspace"""
+from __future__ import unicode_literals
+
+from jsonschema import validate
+from jsonschema.exceptions import ValidationError
+
+from storage.configuration.exceptions import InvalidWorkspaceConfiguration
+
+DEFAULT_VERSION = '1.0'
+
+WORKSPACE_CONFIGURATION_SCHEMA = {
+    'type': 'object',
+    'required': ['broker'],
+    'additionalProperties': False,
+    'properties': {
+        'version': {
+            'description': 'Version of the Workspace configuration schema',
+            'type': 'string',
+            'pattern': '^.{0,50}$',
+        },
+        'broker': {
+            'type': 'object',
+            'required': ['type', 'mount'],
+            'additionalProperties': False,
+            'properties': {
+                'type': {
+                    'type': 'string',
+                    'enum': ['nfs'],
+                },
+                'mount': {
+                    'type': 'string',
+                    'minLength': 1,
+                }
+            }
+        }
+    }
+}
+
+
+class WorkspaceConfiguration(object):
+    """Represents the configuration for a storage Workspace.
+    The configuration includes details about the storage broker system required to read, write, move, or delete files
+    within the workspace.
+    """
+
+    def __init__(self, configuration):
+        """Creates a Workspace configuration object from the given dictionary.
+        The general format is checked for correctness, but the specified broker is not checked.
+
+        :param configuration: The Workspace configuration
+        :type configuration: dict
+        :raises InvalidWorkspaceConfiguration: If the given configuration is invalid
+        """
+
+        self._configuration = configuration
+
+        try:
+            validate(configuration, WORKSPACE_CONFIGURATION_SCHEMA)
+        except ValidationError as ex:
+            raise InvalidWorkspaceConfiguration('Invalid Workspace configuration: %s' % unicode(ex))
+
```

Comment:
Need to create the broker with the configured type and call validation method here. Right now the implemented host broker just has an empty implementation for validation. There is a factory used in the workspace code that shows you how to create a broker from its type.